google/cloud/storage/BUILD.bazel

```diff
@@ -46,7 +46,7 @@ cc_library(
     srcs = google_cloud_cpp_storage_grpc_srcs,
     hdrs = google_cloud_cpp_storage_grpc_hdrs,
     defines = ["GOOGLE_CLOUD_CPP_STORAGE_HAVE_GRPC"],
-    local_defines = select({
+    local_defines = ["GOOGLE_CLOUD_CPP_GRPC_HAS_OTEL_PLUGIN"] + select({
```

Comment:
Just curious, now that we are making this thing optional for CMake, will you do the same for Bazel?

If you go for the opt-out flag, the `deps` on this target will need to change. (That might be complicated enough to warrant separate CMake and Bazel PRs)